### PR TITLE
Drop support for Ruby 1.9.3

### LINKED ISSUE #343: Drop support for Ruby 1.9
As we decided in #338, continuing to support Ruby 1.9 might hamstring us in the future. We are going to take this opportunity to drop support for Ruby 1.9, which has reached its end of life.

### PR DESCRIPTION
**Pull request by tfausak**
Fixes #343.

### CODE REVIEW

**Comment by tfausak** [comment]
We never officially supported JRuby or Rubinius. Should we remove them from the Travis build matrix?

**Comment by AaronLasseigne** [comment]
I'd like to continue supporting JRuby. It's a reasonable and used alternative to MRI. Rubinius I could live withouth.

**Comment by tfausak** [comment]
Can you figure out how to get JRuby running in Ruby 2.0 mode on Travis then?

**Comment by AaronLasseigne** [comment]
Sure, I'll look into it.

**Comment by tfausak** [comment]
Might be as easy as [`JRUBY_OPTS='--2.0'](http://stackoverflow.com/a/15281071/1274282).

**Comment by tfausak** [comment]
Well that was unexpected:

```
ERROR:  While executing gem ... (ArgumentError)
    marshal data too short
```

https://travis-ci.org/orgsync/active_interaction/jobs/101112443

File: lib/active_interaction/backports.rb [hunk 1]
```diff
@@ -4,18 +4,6 @@
...
-    def [](name)
-      send(name)
-    end unless method_defined?(:[])
-
-    # Required for Ruby <= 1.9.3.
-    def []=(name, value)
-      send("#{name}=", value)
-    end unless method_defined?(:[]=)
-  end
-
```
Comment (AaronLasseigne) [inline lib/active_interaction/backports.rb]: I thought the plan was to avoid backward incompatible changes for now.

Comment (AaronLasseigne) [inline lib/active_interaction/backports.rb]: As a note I'm fine with dropping 1.9.3 like a bad habit.

Comment (tfausak) [inline lib/active_interaction/backports.rb]: I planned on leaving this in. When I changed the gem spec to require Ruby `>= 2`, I realized that this served absolutely no purpose. If we wanted to continue to unofficially support Ruby 1.9.3, we'd need to keep Ruby `>= 1.9.3` in the gem spec. And that basically means officially supporting Ruby 1.9.3, which defeats the whole point.

All that to say, I've changed my mind. I'm fine with dropping support for 1.9.3. If someone wants to use ActiveInteraction on 1.9.3, they need to use some version < 3.

Comment (AaronLasseigne) [inline lib/active_interaction/backports.rb]: ok

**Comment by AaronLasseigne** [comment]
:shipit:

**Pull request merged**